packages/botonic-core/src/core-bot.ts

```diff
@@ -81,61 +81,33 @@ export class CoreBot {
     session.__locale = locale
   }
 
-  async input({ input, session, lastRoutePath }: any): Promise<BotResponse> {
-    const output = await this.runInput({ input, session, lastRoutePath })
-    if (!session.is_test_integration) {
-      return output
-    }
-    if (!session._botonic_action?.startsWith('create_test_integration_case:')) {
-      return output
-    }
-
-    const [_, onFinishPayloadId] = session._botonic_action.split(
-      'create_test_integration_case:'
-    )
-
-    const newInput = {
-      ...input,
-      payload: onFinishPayloadId,
-      type: INPUT.POSTBACK,
-      data: undefined,
-      text: undefined,
-    }
-
-    session._botonic_action = undefined
-
-    const newOutput = (this.router as Router).processInput(
-      newInput,
-      output.session,
-      output.lastRoutePath
-    )
-
-    const newRequest = {
+  createRequestFromOutput({ input, session, lastRoutePath }, output) {
```

Comment:
Now that we have this function, it would be a good time to type this BotRequest and use this type in all the points of the bot (pre and post functions of the plugin, routes and botonicInit of the actions). 
Maybe it's better to do it in another PR to avoid mixing concepts